org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2018, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.generator;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.xml.type.XMLTypePackage;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.XtextStandaloneSetup;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.testing.GlobalRegistries;
+import org.eclipse.xtext.tests.AbstractXtextTests;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xtext.generator.grammarAccess.GrammarAccessExtensions;
+import org.eclipse.xtext.xtext.generator.parser.antlr.KeywordHelper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Christian Dietrich - Initial contribution and API
+ */
+public class KeywordHelperTest extends AbstractXtextTests {
+	private GlobalRegistries.GlobalStateMemento globalStateMemento;
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		globalStateMemento = GlobalRegistries.makeCopyOfGlobalState();
+		super.setUp();
+		EPackage.Registry.INSTANCE.put(XMLTypePackage.eNS_URI, XMLTypePackage.eINSTANCE);
+		with(XtextStandaloneSetup.class);
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+		globalStateMemento.restoreGlobalState();
```

Comment:
Hmm should we also undo this?
`EPackage.Registry.INSTANCE.put(XMLTypePackage.eNS_URI, XMLTypePackage.eINSTANCE);`

Reply:
restoreGlobalState